$NetBSD: patch-aa,v 1.1.1.1 2008/10/13 22:45:03 bjs Exp $

Support destdir installation, and use ocamlfind to 
set ${libdir} properly.

--- Makefile.orig	2008-09-12 07:58:19.000000000 -0400
+++ Makefile
@@ -30,10 +30,10 @@ endif
 
 bindir          := ${PREFIX}/bin
 docdir		:= ${PREFIX}/share/doc/menhir
-libdir	        := ${PREFIX}/share/menhir
-mandir          := ${PREFIX}/share/man/man1
+libdir	        := `ocamlfind printconf destdir`
+mandir          := ${PREFIX}/${PKGMANDIR}/man1
 MANS            := menhir.1
-DOCS            := manual.pdf demos
+DOCS            := manual.pdf
 MLYLIB          := src/standard.mly
 MENHIRLIB       := menhirLib.cmi menhirLib.cmo menhirLib.cmx menhirLib.o
 
@@ -62,20 +62,22 @@ src/installation.ml:
 # Installation.
 
 install: src/menhir
-	mkdir -p $(bindir)
-	mkdir -p $(libdir)
-	mkdir -p $(docdir)
-	mkdir -p $(mandir)
-	install src/menhir $(bindir)
-	install -m 644 $(MLYLIB) $(libdir)
-	cp -r $(DOCS) $(docdir)
-	cp -r $(MANS) $(mandir)
+	$(BSD_INSTALL_PROGRAM_DIR) $(DESTDIR)$(bindir)
+	$(BSD_INSTALL_LIB_DIR) $(DESTDIR)$(libdir)
+	$(BSD_INSTALL_DATA_DIR) $(DESTDIR)$(docdir)
+	$(BSD_INSTALL_MAN_DIR) $(DESTDIR)$(mandir)
+	install src/menhir $(DESTDIR)$(bindir)
+	$(BSD_INSTALL_LIB) $(MLYLIB) $(DESTDIR)$(libdir)
+	$(BSD_INSTALL_DATA_DIR) $(DESTDIR)${EGDIR}
+	$(BSD_INSTALL_DATA) $(DOCS) $(DESTDIR)$(docdir)
+	$(BSD_INSTALL_MAN) $(MANS) $(DESTDIR)$(mandir)
+	cp -r demos $(DESTDIR)${EGDIR}
 	@cd src && if $(USE_OCAMLFIND) ; then \
 	  echo Installing MenhirLib via ocamlfind. ; \
-	  ocamlfind install menhirLib META $(MENHIRLIB) ; \
+	  ocamlfind install -destdir $(DESTDIR)$(libdir) menhirLib META $(MENHIRLIB) ; \
 	else \
 	  echo Installing MenhirLib manually. ; \
-	  install -m 644 $(MENHIRLIB) $(libdir) ; \
+	  install -m 644 $(MENHIRLIB) $(DESTDIR)$(libdir) ; \
 	fi
 
 uninstall:
